pahfit/model.py

```diff
@@ -387,30 +399,49 @@ def fit(
         # check if observed spectrum is compatible with instrument model
         instrument.check_range([min(x), max(x)], inst)
 
-        # weigths
-        w = 1.0 / uncz
-
-        # construct model and perform fit
-        astropy_model = self._construct_astropy_model(inst, z, use_instrument_fwhm)
-        fit = LevMarLSQFitter(calc_uncertainties=True)
-        self.astropy_result = fit(
-            astropy_model,
-            xz,
-            yz,
-            weights=w,
-            maxiter=maxiter,
-            epsilon=1e-10,
-            acc=1e-10,
+        self.fitter = self._set_up_fitter(
+            inst, z, x=x, use_instrument_fwhm=use_instrument_fwhm
         )
-        self.fit_info = fit.fit_info
+        self.fitter.fit(xz, yz, uncz, maxiter=maxiter)
+
+        # copy the fit results to the features table
+        self._ingest_fit_result_to_features(self.fitter)
+
         if verbose:
-            print(fit.fit_info["message"])
+            print(self.fitter.message)
 
-        self._parse_astropy_result(self.astropy_result)
+    def _ingest_fit_result_to_features(self, fitter: Fitter):
+        """Copy the results from a Fitter to the features table
 
-    def info(self):
-        """Print out the last fit results."""
-        print(self.astropy_result)
+        This is a utility method, executed only at the end of fit(),
+        where the Fitter is passed after Fitter.fit() has been applied.
+        Passing a different Fitter object could be useful for
+        testing.
+
+        """
+        # iterate over the list stored in fitter, so we only get
+        # components that were set up by _set_up_fitter. Having an
+        # ENABLED/DISABLED flag for every feature would be a nice
+        # alternative (and clear for the user).
```

Comment:
Since we spoon-fed, we could store that list, why ask `Fitter` to do so?

Reply:
I do agree about a `disable` flag; I think we'll need to go there at some point.

Reply:
I now store this list as `self.enabled_features`.